dig.go

```diff
@@ -377,7 +377,16 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 			return err
 		}
 	}
-	c.dg.Add(n.paramList.DotNodes(), n.resultList.DotNodes())
+
+	constructor := &dot.Ctor{
+		Name:    n.location.Name,
+		Package: n.location.Package,
+		File:    n.location.File,
+		Line:    n.location.Line,
+		Params:  n.paramList.DotNodes(),
+		Results: n.resultList.DotNodes(),
```

Comment:
minor readability thing: I would create a `newDotCtor(*node)` function in this package so that this code can be,

```
c.dg.Ctors = append(c.dg.Ctors, newDotCtor(n))
```